src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java

```diff
@@ -122,11 +120,11 @@ public boolean checkEnclaveSignature(ResultModel model, String walletAddress) {
             return false;
         }
         final String chainTaskId = model.getChainTaskId();
-        final String wpAuthKey = String.join("-", chainTaskId, walletAddress);
         final String resultHash = HashUtils.concatenateAndHash(chainTaskId, model.getDeterministHash());
         final String resultSeal = HashUtils.concatenateAndHash(walletAddress, chainTaskId, model.getDeterministHash());
         final String messageHash = HashUtils.concatenateAndHash(resultHash, resultSeal);
-        final WorkerpoolAuthorization workerpoolAuthorization = workerpoolAuthorizations.get(wpAuthKey);
+        final Authorization workerpoolAuthorization = authorizationRepository.findByChainTaskIdAndWorkerWallet(
+                chainTaskId, walletAddress).orElse(null);
```

Comment:
What about this format?
```suggestion
        final Authorization workerpoolAuthorization = authorizationRepository
                .findByChainTaskIdAndWorkerWallet(chainTaskId, walletAddress)
                .orElse(null);
```